test/test-execute/exec-supplementarygroups-single-group-user.service

```diff
@@ -2,7 +2,7 @@
 Description=Test for Supplementary Group with only one group and uid 1
 
 [Service]
-ExecStart=/bin/sh -x -c 'test "$$(id -G)" = "1" && test "$$(id -g)" = "1" && test "$$(id -u)" = "1"'
+ExecStart=/bin/sh -x -c 'test "$$(id -g)" = "1" && test "$$(id -u)" = "1" && ! id -G | grep 0'
```

Comment:
@evverx for the test yep it may fail that's why we should not tweak it that much. I tried to write something complex but not enough time.

> I  think we should remove all user/group tests from the test-execute. I'm not sure we can reliable test this on the host. But we can patch the user database in QEMU/nspawn-tests

Actually these test are fine by them self, unless you manually add uid 1 bin to other groups which is not that common... The purpose of test was to make sure that euid and egid are really dropped and to avoid bugs between namespace setup and getgroups() which have undefined behaviour toward effective gid... and where we really do not want to store it there... So they check a real issue against egid 0 but may fail for another case like yours...

So yes please if there is a reliable best way to test these, then yes just remove them and go for that. Another note, one of tests also cover now the path of dynamic user setups and supplementary groups where the original code path didn't handle that case... so maybe just make sure what you remove.

Thanks closing this one then.